File: DataBindingLiveDataJava/app/src/main/java/com/websarva/wings/android/databindinglivedatajava/ui/MainViewModel.java
package com.websarva.wings.android.databindinglivedatajava.ui;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.websarva.wings.android.databindinglivedatajava.data.model.Rectangle;

public class MainViewModel extends ViewModel {
	MutableLiveData<Rectangle> _liveRectangle = new MutableLiveData<>();
	public MainViewModel() {
		_liveRectangle.setValue(new Rectangle(0, 0));
		generateNewHeight();
		generateNewWidth();
	}

	public void generateNewHeight() {
		Rectangle rectangle = _liveRectangle.getValue();
		int height = (int) (Math.random() * 10) + 1;
		int width = rectangle.getWidth();
		rectangle = new Rectangle(height, width);
		_liveRectangle.setValue(rectangle);
	}

	public void generateNewWidth() {
		Rectangle rectangle = _liveRectangle.getValue();
		int height = rectangle.getHeight();
		int width = (int) (Math.random() * 10) + 1;
		rectangle = new Rectangle(height, width);
		_liveRectangle.setValue(rectangle);
	}

	public MutableLiveData<Rectangle> getLiveRectangle() {
		return _liveRectangle;
	}
}

File: DataBindingLiveDataKotlin/app/src/main/java/com/websarva/wings/android/databindinglivedatakotlin/ui/MainViewModel.kt
package com.websarva.wings.android.databindinglivedatakotlin.ui

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.websarva.wings.android.databindinglivedatakotlin.data.model.Rectangle

class MainViewModel : ViewModel() {
	val liveRectangle: MutableLiveData<Rectangle> = MutableLiveData()

	init {
		liveRectangle.value = Rectangle(0, 0)
		generateNewHeight()
		generateNewWidth()
	}

	fun generateNewHeight() {
		var rectangle = liveRectangle.value
		val height = (Math.random() * 10).toInt() + 1
		val width = rectangle!!.width
		rectangle = Rectangle(height, width)
		liveRectangle.value = rectangle
	}

	fun generateNewWidth() {
		var rectangle = liveRectangle.value
		val height = rectangle!!.height
		val width = (Math.random() * 10).toInt() + 1
		rectangle = Rectangle(height, width)
		liveRectangle.value = rectangle
	}
}
